cpp/src/integrations/cudf/partition.cpp

```diff
@@ -20,6 +21,61 @@
 
 namespace rapidsmpf {
 
+
+namespace {
+
+
+/**
+ * @brief Calculate the memory usage of a column.
+ *
+ * @param col The column to calculate the memory usage of.
+ * @return The memory usage of the column.
+ */
+size_t column_memory_usage(cudf::column_view const& col) {
```

Comment:
What about submitting a small PR that introduces an util function `approx_device_memory_usage(x)` and implements overloads for common data structures like `cudf::table_view`, `cudf::coulmn_view` and vectors/ranges of them?